skimage/feature/blob.py

```diff
@@ -254,29 +275,41 @@ def blob_dog(image, min_sigma=1, max_sigma=50, sigma_ratio=1.6, threshold=2.0,
     gaussian_images = [gaussian_filter(image, s) for s in sigma_list]
 
     # computing difference between two successive Gaussian blurred images
-    # multiplying with standard deviation provides scale invariance
+    # multiplying with average standard deviation provides scale invariance
     dog_images = [(gaussian_images[i] - gaussian_images[i + 1])
-                  * sigma_list[i] for i in range(k)]
+                  * np.mean(sigma_list[i]) for i in range(k)]
 
     image_cube = np.stack(dog_images, axis=-1)
 
     # local_maxima = get_local_maxima(image_cube, threshold)
     local_maxima = peak_local_max(image_cube, threshold_abs=threshold,
                                   footprint=np.ones((3,) * (image.ndim + 1)),
                                   threshold_rel=0.0,
-                                  exclude_border=False)
+                                  exclude_border=exclude_border)
     # Catch no peaks
     if local_maxima.size == 0:
         return np.empty((0, 3))
+
     # Convert local_maxima to float64
     lm = local_maxima.astype(np.float64)
-    # Convert the last index to its corresponding scale value
-    lm[:, -1] = sigma_list[local_maxima[:, -1]]
+
+    # translate final column of lm, which contains the index of the
+    # sigma that produced the maximum intensity value, into the sigma
+    sigmas_of_peaks = sigma_list[local_maxima[:, -1]]
+
+    # if the gaussian is isotropic, the stdev across dimensions are
+    # identical, so return only the stdev deviation of the first dimension
+    if np.unique(min_sigma).shape == (1,) and np.unique(max_sigma).shape == (1,):
+        sigmas_of_peaks = sigmas_of_peaks[:, 0][:, None]
+
+    # Remove sigma index and replace with sigmas
+    lm = np.hstack([lm[:, :-1], sigmas_of_peaks])
+
     return _prune_blobs(lm, overlap)
 
 
 def blob_log(image, min_sigma=1, max_sigma=50, num_sigma=10, threshold=.2,
-             overlap=.5, log_scale=False):
+             overlap=.5, log_scale=False, exclude_border=False):
```

Comment:
```suggestion
             overlap=.5, log_scale=False, *, exclude_border=False):
```

We are using [keyword-only arguments](https://www.python.org/dev/peps/pep-3102/) going forward as a way to get more flexibility when modifying the API in the future.